src/backend/parser/gram.y

```diff
@@ -214,6 +214,9 @@ static Node *makeRecursiveViewSelect(char *relname, List *aliases, Node *query);
 rewrite_typmod_expr_hook_type rewrite_typmod_expr_hook = NULL;
 validate_numeric_typmods_hook_type validate_numeric_typmods_hook = NULL;
 check_recursive_cte_hook_type check_recursive_cte_hook = NULL;
+
+/* Dump and Restore */
+bool restore_tsql_datetime2 = false;
```

Comment:
do we need a separate flag? checking for TSQL restore flag should be sufficient

Reply:
We are using dump_restore flag based on object types, e.g., ITVF, T-SQL views, etc. However, datetime2 flag is needed for all objects having datetime2 columns. So, I think introducing a separate flag would be more safe.

Reply:
As discussed, we will use dump_restore GUC for this case.